src/EFCore.SqlServer/Query/Internal/SqlServerQueryRootCreator.cs

```
@@ -0,0 +1,103 @@
+    ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+    /// </summary>
+    public class SqlServerQueryRootCreator : QueryRootCreator
+    {
+        /// <summary>
+        ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
+        ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
+        ///     any release. You should only use it directly in your code with extreme caution and knowing that
+        ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+        /// </summary>
+        public override QueryRootExpression CreateQueryRoot(IEntityType entityType, QueryRootExpression? source)
+        {
+            if (source is TemporalQueryRootExpression tqre)
+            {
+                if (!entityType.GetRootType().IsTemporal())
+                {
+                    throw new InvalidOperationException(SqlServerStrings.TemporalNavigationExpansionBetweenTemporalAndNonTemporal(entityType.DisplayName()));
+                }
+
+                if (tqre is TemporalAsOfQueryRootExpression asOf)
+                {
+                    return source.QueryProvider != null
+                        ? new TemporalAsOfQueryRootExpression(source.QueryProvider, entityType, asOf.PointInTime)
+                        : new TemporalAsOfQueryRootExpression(entityType, asOf.PointInTime);
+                }
+
+                throw new InvalidOperationException(SqlServerStrings.TemporalNavigationExpansionOnlySupportedForAsOf(nameof(TemporalOperationType.AsOf)));
+
+            }
+
+            if (entityType.GetRootType().IsTemporal()
+                && source is null)
+            {
+                throw new InvalidOperationException(SqlServerStrings.TemporalFailedToCreateQueryRoot(entityType.DisplayName()));
```

Shouldn't this be non-temporal access, i.e. current data.

---

I'd prefer throwing in this case. This should only be relevant in niche cases when the source is not QRE, e.g. defining query (anything else?) It's gonna be easier to find and fix those edge cases if we throw rather than return current data